conkit/plot/contactplotplus.py

```diff
@@ -0,0 +1,267 @@
+                             toolbar_location="above")
+        self.add_contact_layer()
+
+    @property
+    def spectrum(self):
+        return {1: '#f7fbff',
+                2: '#deebf7',
+                3: '#c6dbef',
+                4: '#9ecae1',
+                5: '#6baed6',
+                6: '#4292c6',
+                7: '#2171b5',
+                8: '#08519c',
+                9: '#08306b'
+                }
+
+    @property
+    def aa_properites(self):
+        return {'A': 'Non-polar, aliphatic',
+                'R': 'Positively charged (basic; non-acidic); Polar; Hydrophilic; pK=12.5',
+                'N': 'Polar, non-charged',
+                'D': 'Negatively charged (acidic); Polar; Hydrophilic; pK=3.9',
+                'C': 'Polar, non-charged',
+                'E': 'Negatively charged (acidic); Polar; Hydrophilic; pK=4.2',
+                'Q': 'Polar, non-charged',
+                'G': 'Non-polar, aliphatic',
+                'H': 'Positively charged (basic; non-acidic); Polar; Hydrophilic; pK=6.0',
+                'I': 'Non-polar, aliphatic',
+                'L': 'Non-polar, aliphatic',
+                'K': 'Positively charged (basic; non-acidic); Polar; Hydrophilic; pK=10.5',
+                'M': 'Polar, non-charged',
+                'F': 'Aromatic',
+                'P': 'Non-polar, aliphatic',
+                'S': 'Polar, non-charged',
+                'T': 'Polar, non-charged',
+                'W': 'Aromatic',
+                'Y': 'Aromatic',
+                'V': 'Aromatic'
+                }
+
+    @property
+    def layers(self):
+        return {"Mem_pred-I": [self.mempred_i],
+                "Mem_pred-M": [self.mempred_m],
+                "Mem_pred-O": [self.mempred_o],
+                "SS-H": [self.sspred_h],
+                "SS-C": [self.sspred_c],
+                "SS-E": [self.sspred_e],
+                "Disorder": [self.dispred_dis],
+                "Ordered": [self.dispred_o],
+                "Contact": [self.contact_circles],
+                "Mismatch": [self.contact_w_reference_mis],
+                "Match": [self.contact_w_reference_match],
+                "Reference pdb": [self.reference_points]
+                }
+
+    @property
+    def legend(self):
+
+        # had to change conditional inequality otherwise legend unless all layers present
+        return Legend(items=[(key, value) for key, value in self.layers.items() if value != [None]],
+                      location='center')
+
+    def add_contact_layer(self):
+        id = [x.id for x in self.conpred]
+        res1 = [x.res1 for x in self.conpred]
+        res1_chain = [x.res1_chain for x in self.conpred]
+        res1_seq = [x.res1_seq for x in self.conpred]
+        res2 = [x.res2 for x in self.conpred]
+        res2_chain = [x.res2_chain for x in self.conpred]
+        res2_seq = [x.res2_seq for x in self.conpred]
+        raw_score = [x.raw_score for x in self.conpred]
```

Comment:
why do these need to be unpacked all individually? they come from the same attribute ``self.conpred``.